text/0148-sampling-probability.md

```diff
@@ -0,0 +1,726 @@
+
+#### Adjusted sample count
+
+Following the model above, every event defines the notion of an
+_adjusted count_.
+
+- _Adjusted count_ is zero if the event was not selected for the sample
+- _Adjusted count_ is the reciprocal of its inclusion probability, otherwise.
+
+The adjusted count of an event represents the expected contribution to
+the estimated population total of a sample frame represented by the
+individual event.
+
+The use of a reciprocal inclusion probability matches our intuition
+for probabilities.  Items selected with "one-out-of-N" probability of
+inclusion count for N each, approximately speaking.
+
+This intuition is backed up with statistics.  This equation is known
+as the Horvitz-Thompson estimator of the population total, a
+general-purpose statistical "estimator" that applies to all _without
+replacement_ sampling designs.
+
+Assuming sample data is correctly computed, the consumer of sample
+data can treat every sample event as though an identical copy of
+itself has occurred _adjusted count_ times.  Every sample event is
+representative for adjusted count many copies of itself.
+
+There is one essential requirement for this to work.  The selection
+procedure must be _statistically unbiased_, a term meaning that the
+process is required to give equal consideration to all possible
+outcomes.
+
+#### Sampling and variance
+
+The use of unbiased sampling outlined above makes it possible to
+estimate the population total for arbitrary subsets of the sample, as
+every individual sample has been independently assigned an adjusted
+count.
+
+There is a natural relationship between statistical bias and variance.
+Approximate counting comes with variance, a matter of fact which can
+be controlled for by the sample size.  Variance is unavoidable in an
+unbiased sample, but it vanishes when you have enough data.
+
+Although this makes it sounds like small sample sizes are a problem,
+due to expected high variance, this is just a limitation of the
+technique.  When variance is high, use a larger sample size.
+
+An easy approach for lowering variance is to aggregate sample frames
+together across time.  For example, although the estimates drawn from
+a one-minute sample may have high variance, combining an hour of
+one-minute sample frames into an aggregate data set is guaranteed to
+lower variance.  It must, because the data remains unbiased.
+
+### Conveying the sampling probability
+
+Some possibilities for encoding the adjusted count or inclusion
+probability are discussed below, depending on the circumstances and
+the protocol.
+
+There are several ways of encoding this information:
+
+- as a dedicated field in an OTLP protobuf message
+- as a non-descriptive Attribute in an OTLP Span, Metric, or Log
+- without any dedicated field.
+
+#### Encoding adjusted count
+
+We can encode the adjusted count directly as a floating point or
+integer number in the range [0, +Inf).  This is a conceptually easy
```

Comment:
A uniform sample of 40% has adjusted count of 2.5. To me this, makes sense, but I realize that it can present problems in some systems. This is mentioned in the example about computing Histogram metrics from spans, because OTLP Histogram points do not have a way to represent non-integer counts.  One of the reasons I like using an attribute to reflect adjusted count, not multiplying these values into the data itself, is that it allows the use of floating-point counts even where values are integers; OTOH, it punts the problem for the next consumer. I am happy to recommend the use of integer-reciprocal sampling rates in situations where this is a real problem.

Reply:
I added a paragraph to state the above. ^^^